nightfall-deployer/contracts/State.sol

```diff
@@ -223,6 +215,54 @@ contract State is ReentrancyGuardUpgradeable, Pausable, Config {
                         revert(0, 0)
                     }
                 }
+
+                if iszero(
+                    calldataload(
+                        add(add(t.offset, calldataload(add(t.offset, mul(0x20, i)))), mul(0x20, 1))
+                    )
+                ) {
+                    mstore(x, checkTxEthFeesSignature) //Store the signature of the function in x
+                    mstore(add(x, 0x04), mload(add(transactionHashesPos, mul(0x20, i)))) //Store the transactionHash after the signature
+                    pop(
+                        call(
+                            // Call getTransactionFeesEth function to see if funds has been deposited
+                            gas(),
+                            shieldAddress.slot, //To addr
+                            0, //No value
+                            x, //Inputs are stored at location x
+                            add(0x04, 0x20), //Inputs are 36 bytes long
+                            x, //Store output over input (saves space)
+                            0x20 //Outputs are 32 bytes long
+                        )
+                    )
+
+                    mstore(feePayments, add(mload(x), mload(feePayments)))
+                }
+
+                if eq(
```

Comment:
why is this `eq( isZero (x),0)`  can we just test `isZero`?

Reply:
In this case we want to check that is not zero. So, fees will be paid on chain if the transaction is not a deposit, which means transactionType will be not 0. isZero function is 1 if what is evaluated is zero and 0 otherwise. Since we want to get inside the if we need to check that isZero(x) is equal to 0

Reply:
ahh ok I get it the `isZero` is true (i.e. 1) so the test for not zero is to compare it with 0